tools/cwrap/plugins/Broadcast.py

```diff
@@ -0,0 +1,106 @@
+from . import CWrapPlugin
+from string import Template
+
+
+class Broadcast(CWrapPlugin):
```

Comment:
Adds boilerplate to broadcast arguments to be identical sizes, before running the algorithm proper. Only works for binary operators.